### ISSUE TITLE
error train on car re-id model 

### ISSUE THREAD

**Issue by qiyayun** [VisualComputingInstitute/triplet-reid#65]
could you tell me your cars196  args.json
this is my args.json
{
  "batch_k": 4,
  "batch_p": 32,
  "checkpoint_frequency": 800,
  "crop_augment": false,
  "decay_start_iteration": 2000,
  "detailed_logs": false,
  "embedding_dim": 128,
  "experiment_root": "/home/dip_image/my_experiment",
  "flip_augment": false,
  "head_name": "fc1024",
  "image_root": "data/",
  "initial_checkpoint": "checkpoint/resnet_v1_50.ckpt",
  "learning_rate": 0.0003,
  "loading_threads": 8,
  "loss": "batch_hard",
  "margin": "soft",
  "metric": "euclidean",
  "model_name": "resnet_v1_50",
  "net_input_height": 256,
  "net_input_width": 128,
  "pre_crop_height": 288,
  "pre_crop_width": 144,
  "resume": false,
  "train_iterations": 4000,
  "train_set": "data/market1501_train.csv"
}

**Comment by qiyayun** [comment]
If I use the car datasets, how should I evaluate the MAP( how to build query.csv in CARS196)thanks a lot

**Comment by Pandoro** [comment]
```
{
  "batch_k": 4,
  "batch_p": 18,
  "checkpoint_frequency": 200,
  "crop_augment": true,
  "decay_start_iteration": 4000,
  "embedding_dim": 128,
  "experiment_root": "/exp/dir/reid/cars-final/defense-long",
  "flip_augment": true,
  "head_name": "direct",
  "image_root": "/data/cars196/",
  "initial_checkpoint": "/data/tf-pretrained/resnet_v1_50.ckpt",
  "learning_rate": 0.0003,
  "loading_threads": 8,
  "loss": "batch_hard",
  "margin": 0.1,
  "metric": "euclidean",
  "model_name": "resnet_v1_50",
  "net_input_height": 224,
  "net_input_width": 224,
  "optimizer": "AdamOptimizer(1e-4, epsilon=1e-4)",
  "pre_crop_height": 256,
  "pre_crop_longer_side": null,
  "pre_crop_shorter_side": null,
  "pre_crop_width": 256,
  "resume": false,
  "train_iterations": 8000,
  "train_set": "data/cars196_train.csv"
}
```
This is one of the `args.json` we used, as you can see this is not directly compatible with the code released since for example the "optimizer" argument as such doesn't exist in the public code. But essentially that is just what is passed to an `eval` as a quick hack and thus it never made it to the public code.

Important differences I see are different PK-batches, a more sensible image size, a different learning schedule and a different learning rate. As we pointed out, it is super crucial that you tune your learning rate. This is probably one of your big issues since you are using the Market-1501 learning rate.

All of this is fairly long ago though, so you will just have to play around a little with how exactly you train it, i.e. which values for `P`, `K` and the learning rate as most obvious starting points.

**Comment by Pandoro** [comment]
In that case you should evaluate on the test set from cars196, or even better if you want to do it really correct, create a train/validation split to perform hyper parameter tuning.

**Comment by Pandoro** [comment]
I don't know what predict.txt or matlab code you mean. You have to look at the excluders. Please try to actually understand what is going on in the code and how the evaluation works. Otherwise there is no point in doing an evaluation at all if you don't know what is going on.

**Comment by qiyayun** [comment]
Thanks for your reply a lot ,I have found the the way in your code .
